compiler/src/iree/compiler/Preprocessing/TransformExtensions/PreprocessingExtensions.cpp

```diff
@@ -223,6 +223,114 @@ IREE::transform_dialect::MatchCastCompatibleDagFromRootOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// MatchContractionOp
+//===----------------------------------------------------------------------===//
+
+DiagnosedSilenceableFailure
+IREE::transform_dialect::MatchContractionOp::matchOperation(
+    Operation *current, transform::TransformResults &results,
+    transform::TransformState &state) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(current);
+  if (!linalgOp) {
+    return emitSilenceableFailure(current->getLoc())
+           << "Operation " << *current << " is not a LinalgOp.";
+  }
+
+  if (!linalg::isaContractionOpInterface(linalgOp)) {
+    return emitSilenceableFailure(current->getLoc())
+           << "Operation " << *current << " is not a contraction operation.";
+  }
+
+  if (std::optional<ArrayAttr> indexingMaps = getIndexingMaps()) {
+    ArrayAttr currentIndexingMaps = linalgOp.getIndexingMaps();
+    ArrayAttr targetIndexingMaps = *indexingMaps;
+    if (currentIndexingMaps.size() != targetIndexingMaps.size()) {
```

Comment:
Why not compare two array attributes? I don't think there's value in producing verbose silenceable failures -- nobody looks at them